sympy/ntheory/generate.py

```diff
@@ -302,12 +303,27 @@ def __contains__(self, n):
         a, b = self.search(n)
         return a == b
 
+    def __iter__(self):
+        for n in count(1):
+            yield self[n]
+
     def __getitem__(self, n):
         """Return the nth prime number"""
         if isinstance(n, slice):
             self.extend_to_no(n.stop)
-            return self._list[n.start - 1:n.stop - 1:n.step]
+            # Python 2.7 slices have 0 instaed of None for start, so
```

Comment:
Where does this information come from? Python 2.7 [manual](https://docs.python.org/2.7/library/functions.html#slice) claims
> The start and step arguments default to None.

Reply:
That's strange, IIRC I checked it myself after a pytest error on start being equal to 0, let me double check it.

Reply:
OK so the doc is right, and the 0 is also here, so it appears probably in the implementation of `[]`, I don't want to check the code, but here's my test:

In Python 2.7.16 :
```
>>> slice(10)
slice(None, 10, None)  # As the doc tells
>>> class A:
...     def __getitem__(self, item):
...         return item
... 
>>> A()[:10]
slice(0, 10, None)
>>> A()[:10] == A()[0:10]
True  # We can't distinguish between `[:10]` and `[0:10]` in Python 2.7.
```